tests/check_non_terminating.sh

```diff
@@ -0,0 +1,166 @@
+}
+
+getFirstContainerComponentCommand() {
+    devfilePath=$1
+    local _gfccc_command=()
+    local _gfccc_command_string=()
+
+    IFS=" " read -r -a _gfccc_command_string <<< "$($YQ_PATH eval '[ .components[] | select(has("container")) ] | .[0].container.command[]? + " "' "$devfilePath" -r | paste -s -d '\0' -)"
+    if (( ${#_gfccc_command_string[@]} == 0 )); then
+        echo ""
+    else
+        for command_word in "${_gfccc_command_string[@]}"; do
+            _gfccc_command+=("${command_word}")
+        done
+        echo "${_gfccc_command[@]}"
+    fi
+}
+
+getFirstContainerComponentArgs() {
+    devfilePath=$1
+    local _gfcca_args=()
+    local _gfcca_args_string=()
+
+    IFS=" " read -r -a _gfcca_args_string <<< "$($YQ_PATH eval '[ .components[] | select(has("container")) ] | .[0].container.args[]? + " "' "$devfilePath" -r | paste -s -d '\0' -)"
+    if (( ${#_gfcca_args_string[@]} == 0 )); then
+      echo ""
+    else
+        for arg in "${_gfcca_args_string[@]}"; do 
+            _gfcca_args+=("${arg}")
+        done
+        echo "${_gfcca_args[@]}"
+    fi
+}
+
+isNonTerminating() {
+    _int_image=$1
+    _int_command=("$2")
+    _int_command_args=("$3")
+
+    namespace=default
+    timeout_in_sec=60 # <== includes image pulling
+
+    echo "  PARAMS: image --> $_int_image, command --> ${_int_command[*]}, args --> ${_int_command_args[*]}"
+ 
+    if [ "${_int_command[*]}" == "null" ] && [ "${_int_command_args[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image"
+    elif [ "${_int_command[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image -- ${_int_command_args[*]}\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" -- ${_int_command_args[*]}
+    elif [ "${_int_command_args[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image --command -- ${_int_command[*]}\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" --command=true -- ${_int_command[*]}
+    else
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image --command -- ${_int_command[*]} ${_int_command_args[*]}\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" --command=true -- ${_int_command[*]} ${_int_command_args[*]}
+    fi
+    
+    if 2>/dev/null 1>/dev/null kubectl wait pods -n ${namespace} test-terminating --for condition=Ready --timeout=${timeout_in_sec}s; then
```

Comment:
The run commands hard-code the namespace (`default`) whereas the variable is used here.

Reply:
Thanks, I will fix that